varname.py

```diff
@@ -19,18 +19,39 @@
 class VarnameRetrievingError(Exception):
     """When failed to retrieve the varname"""
 
-def varname(caller: int = 1, raise_exc: bool = True) -> Optional[str]:
+def varname(caller: int = 1,
+            nvars: int = 1,
+            raise_exc: bool = True) -> Optional[Union[str, Tuple[str]]]:
     """Get the variable name that assigned by function/class calls
 
     Args:
         caller: The call stack index, indicating where this function
             is called relative to where the variable is finally retrieved
+        nvars: Number of variables to be expected on left-hand side (LHS).
+            if `nvars` > 1 or `nvars` is `None`, a tuple of variable names
+            will be returned.
+            Note that for `nvars` is `None`, this function will return
+            a tuple even when there is only one variable on the LHS
+            For example:
+            >>> def func():
+            >>>   return varname(nvars=2)
+            >>> a, b = func()
+            >>> # (a, b) == ('a', 'b')
+            >>>
+            >>> def func():
+            >>>   return varname(nvars=None)
+            >>> a = func() # a == ('a', )
+
+            if nvars == 1, a single variable name will be returned.
+            if LHS is hierarchical, this denotes the number of first-level
```

Comment:
@alexmojaki Do you think we should also allow `nvars` to match the hierarchical LHS?
For example, `nvars=(1,2)` makes it match `a, (b, c)`, and `nvars=(1, (2, 3))` matches `a, ((b, c), (d, e, f))`.
But this will definitely make it complicated.

Reply:
I think it's too complicated. Users can do these kinds of checks themselves.

In fact, thinking about this is making me realize that it's already too complicated. Currently there's three modes of behaviour, depending on whether `nvars` is `None`, `1`, or `>1`. Users only really need two: return a single name (only one name is allowed) or a tuple (anything is allowed). The only difference between `None` and `>1` is that we check the number of variables. But that's a really simple operation that users can do themselves. The option just makes the API and documentation more complicated and creates potential compatibility problems down the line.

I suggest changing `nvars` to a boolean flag called `allow_multiple` or `return_tuple` or something.

Reply:
Sounds reasonable.